pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -312,28 +312,43 @@ func (n *networkManager) extractGatewaysInner(svc *model.Service) bool {
 
 // getGatewayDetails returns gateways without the address populated, only the network and (unmapped) port for a given service.
 func (n *networkManager) getGatewayDetails(svc *model.Service) []model.NetworkGateway {
-	// TODO should we start checking if svc's Ports contain the gateway port?
-
 	// We have different types of E/W gateways - those that use mTLS (those are used in sidecar mode when talking cross networks)
 	// and those that use double-HBONE (those are used in ambient mode when talking cross cluster). A gateway service may or may
 	// not listen on the mTLS (15443, by default) or HBONE (15008) ports, depending on the mode of operation used by the mesh
 	// in the remote cluster. We should not use gateways that don't really listen on the right port.
-	hbonePort := uint32(DefaultNetworkGatewayHBONEPort)
-	if _, exists := svc.Ports.GetByPort(DefaultNetworkGatewayHBONEPort); !exists {
-		hbonePort = 0
-	}
 
 	// label based gateways
 	// TODO label based gateways could support being the gateway for multiple networks
 	if nw := svc.Attributes.Labels[label.TopologyNetwork.Name]; nw != "" {
+		hbonePort := DefaultNetworkGatewayHBONEPort
```

Comment:
This isn't specific to this PR, but one assumption that week make with ambient multi-network is that the e/w gateway can only be reached on port 15008. There may be reasons an e/w gateway may want to receive double hbone requests on different ports (public vs. private port or multitenancy), so we should consider documenting these assumptions in the form of TODOs (not a requirement)